third_party/utils/src/test/io/bazel/rulesscala/utils/Scala3CompilerUtils.scala

```diff
@@ -4,27 +4,76 @@ import dotty.tools.dotc.Compiler
 import dotty.tools.dotc.core.Contexts._
 import dotty.tools.dotc.reporting.{Diagnostic, StoreReporter}
 import dotty.tools.dotc.util.SourceFile
+import dotty.tools.io.{AbstractFile, VirtualDirectory}
 
-import java.nio.file.Paths
+import java.nio.file.{Path, Paths}
+import scala.util.control.NonFatal
 
 object Scala3CompilerUtils {
+  sealed trait CompileResult {
```

Comment:
This looks a bit suspicious. You have an ADT to represent success and failure and also have a boolean field for same purpose. Could `isSuccess` be removed?